static/js/nearYouOverlay.js

```diff
@@ -33,9 +41,120 @@ var Overlay = L.Class.extend({
     },
 
+    addDesktopOpenButton: function () {
+        var self = this;
+
+        // Create a button to open/close the sidebar
+        let _desktopOpenButton = document.createElement('button');
+        _desktopOpenButton.classList.add('desktop');
+        _desktopOpenButton.id = 'nearYouDesktopToggleButton';
+        _desktopOpenButton.innerHTML = 'Show Stickers';
+        document.body.appendChild(_desktopOpenButton);
+
+        _desktopOpenButton.addEventListener('click', function() {
+            if (self.isActive === false) {
+                self.openDesktopSidebar();
+            } else {
+                self.closeDesktopSidebar();
+            }
+        });
+
+        return _desktopOpenButton;
+    },
+
+    switchToMobile: function () {
+        this.currentlyMobile = true;
+        this.closeDesktopSidebar();
+        this._overlayElement.classList.remove('desktop');
+        this._overlayElement.classList.add('mobile');
+        this._desktopOpenButton.classList.remove('desktop');
+        this._desktopOpenButton.classList.add('mobile');
+
+        this.addTouchEventListeners();
+    },
+
+    switchToDesktop: function () {
+        this.currentlyMobile = false;
+        this.closeMobileOverlay();
+        this._overlayElement.classList.remove('mobile');
+        this._overlayElement.classList.add('desktop');
+        this._desktopOpenButton.classList.remove('mobile');
+        this._desktopOpenButton.classList.add('desktop');
+
+        this._overlayElement.style = "";
+
+        this.removeTouchEventListeners();
+    },
+
+    openMobileOverlay: function () {
+        this._overlayElement.classList.add('active');
+        this._overlayElement.classList.remove('inactive');
+        this._desktopOpenButton.classList.add('active');
+        this._desktopOpenButton.classList.remove('inactive');
+
+        this._desktopOpenButton.textContent = "Hide Stickers";
+
+        // This styling needs to be in the javascript because it needs to override the onTouchMove function
+        this._overlayElement.style.bottom = "0";
+
+        if (!this.isActive) {
+            this.getNearYouData();
+        }
+        
+        this.isActive = true;
+    },
+
+    closeMobileOverlay: function () {
+        this._overlayElement.classList.add('inactive');
+        this._overlayElement.classList.remove('active');
+        this._desktopOpenButton.classList.add('inactive');
+        this._desktopOpenButton.classList.remove('active');
+
+        this._desktopOpenButton.textContent = "Show Stickers";
+
+        this.isActive = false;
+
+        // This styling needs to be in the javascript because it needs to override the onTouchMove function
+        this._overlayElement.style.bottom = "-90%";
+
+        this._overlayElement.scrollTo(0, 0);
+        let stickerDivs = document.querySelectorAll('.stickerDiv');
+        stickerDivs.forEach(div => div.classList.remove('revealed'));
+    },
+
+    openDesktopSidebar: function () {
```

Comment:
You have quite some repetitive code for the mobile and desktop open and close and such. Perhaps it is better to abstract the real logic in a method, and give parameters such as `mobile, open`.

Reply:
I rewrote the toggling of the overlay to be more readable and clean. I don't think there is much more to optimize in terms of repetitive code, so I'll resolve this conversation